isisd/isis_zebra.c

```diff
@@ -549,24 +593,38 @@ void isis_zebra_release_dynamic_label(mpls_label_t label)
 	}
 }
 
+/**
+ * Connect to the Label Manager.
+ */
 static void isis_zebra_label_manager_connect(void)
 {
 	/* Connect to label manager. */
 	while (zclient_socket_connect(zclient_sync) < 0) {
-		zlog_warn("%s: error connecting synchronous zclient!",
+		zlog_warn("%s: re-attempt connecting synchronous zclient!",
 			  __func__);
 		sleep(1);
 	}
+	/* make socket non-blocking */
 	set_nonblocking(zclient_sync->sock);
+
+	/* Send hello to notify zebra this is a synchronous client */
+	while (zclient_send_hello(zclient_sync) < 0) {
+		zlog_warn(
+			"%s: re-attempt sending hello for synchronous zclient!",
+			__func__);
+		sleep(1);
```

Comment:
Doing this means we don't respond to vtysh or watchfrr pings, which can result in watchfrr assuming isisd has died, with bad consequences. A thread timer would be better.

Reply:
@qlyoung Agree, but we need to also update ldpd and ospfd accordingly. Can you provide a example of thread timer ?

Reply:
I've opened PR #6274  with an example for ldpd. But really, just because ldpd does a bad thing doesn't mean that isis can't do better, eh?

Reply:
@odd22 I'm referring to `thread_add_timer`